core/src/main/java/hudson/model/ParametersAction.java

```diff
@@ -88,9 +89,27 @@
     private transient AbstractBuild<?, ?> build;
 
     private transient Run<?, ?> run;
+    /**
+     * Just a temporary variable for diagnostics of the root cause of JENKINS-39495.
+     * This field is being used in {@link #onLoad(hudson.model.Run)} and {@link #onAttached(hudson.model.Run)} hooks, which provide info about the run.
+     */
+    private transient boolean wasInitializedWithNullParameters;
 
-    public ParametersAction(List<ParameterValue> parameters) {
-        this.parameters = parameters;
+    /**
+     * Constructs a new action with a specified list of parameter values.
+     * @param parameters Parameter values.               
+     */
+    @SuppressWarnings("null")
+    public ParametersAction(@Nonnull List<ParameterValue> parameters) {
+        wasInitializedWithNullParameters = parameters == null;
+        if (wasInitializedWithNullParameters && LOGGER.isLoggable(Level.FINE)) {
+            LOGGER.log(Level.FINE, "Parameters action has been initialized with null parameters. "
+                    + "It is a bug in a plugin or script, which created this action (see JENKINS-39495). " 
+                    + "See next system log entries to see for which Run it has been created",
+                    new IllegalStateException("Parameters list is null for the action"));
```

Comment:
Well there are plenty of cases where something causes a similar exception to recur. We still want to see the exception. Admins encountering it may be motivated to file a bug report for it; and we can improve Jenkins core and `support-core` to deduplicate logs.